src/tests/end-to-end/utils/utils.cpp

```diff
@@ -4,6 +4,208 @@
 
 using namespace Antares::Data;
 
+
+void initializeStudy(Study::Ptr study)
+{
+    study->resultWriter = std::make_shared<NullResultWriter>();
+    study->parameters.reset();
+    Data::Study::Current::Set(study);
+}
+
+void setNumberMCyears(Study::Ptr study, unsigned int nbYears)
+{
+    study->parameters.resetPlaylist(nbYears);
+    study->bindingConstraints.resizeAllTimeseriesNumbers(nbYears);
+}
+
+void configureLinkCapacities(AreaLink* link)
+{
+    const double linkCapacityInfinite = +std::numeric_limits<double>::infinity();
+    link->directCapacities.resize(1, 8760);
+    link->directCapacities.fill(linkCapacityInfinite);
+
+    link->indirectCapacities.resize(1, 8760);
+    link->indirectCapacities.fill(linkCapacityInfinite);
+}
+
+std::shared_ptr<ThermalCluster> addClusterToArea(Area* area, const std::string& clusterName)
+{
+    auto cluster = std::make_shared<ThermalCluster>(area);
+    cluster->setName(clusterName);
+    cluster->reset();
+
+    auto added = area->thermal.list.add(cluster);
+
+    area->thermal.list.mapping[cluster->id()] = added;
+    area->thermal.prepareAreaWideIndexes();
+
+    return cluster;
+}
```

Comment:
Move to main code ? There seems to be code duplication here

Reply:
Sorry I miss your point, could you be a bit more specific ? which code duplication ?

Reply:
This piece of code is a copy-paste of `ThermalClusterList::loadFromFolder`. Maybe we could write a function in libs/antares/study to factorize this.